go/private/context.bzl

```diff
@@ -241,6 +241,7 @@ def go_context(ctx, attr = None):
     env.update({
         "GOARCH": mode.goarch,
         "GOOS": mode.goos,
+        "GOCACHE": "off", # bazel has it's own caching
```

Comment:
As was mentioned in another comment, this will probably break in Go 1.12. I think they want to integrate the cache more tightly into `go build`.

`GoStdLib` should be the only action that needs this. The other actions all read inputs and write outputs directly. So maybe `GOCACHE` could just be set in stdlib.go?

I share your concern about interacting with the user's cache outside of Bazel's workspace. It's probably okay as long as sandboxing or remote execution are in use, but that's not guaranteed. It's probably better to create a temp directory, set `GOCACHE` to that, then delete it at the end of the action.

It's probably best for all that to be in a separate PR though. Or feel free to file an issue, and I can fix it next week.

Reply:
sure thing, I'll remove it